test/util/testnode/read.go

```diff
@@ -39,6 +39,63 @@ func ReadBlockchain(ctx context.Context, rpcAddress string) ([]*types.Block, err
 	return ReadBlockHeights(ctx, rpcAddress, 1, status.SyncInfo.LatestBlockHeight)
 }
 
+// ReadBlockchainHeaders retrieves the blockchain headers from height 1 up to
+// latest available height from the node at rpcAddress and returns it.
+// The headers are returned in ascending order (lowest first).
+func ReadBlockchainHeaders(ctx context.Context, rpcAddress string) ([]*types.BlockMeta, error) {
+	client, err := http.New(rpcAddress, "/websocket")
+	if err != nil {
+		return nil, err
+	}
+
+	// fetch the latest height
+	resp, err := client.Status(ctx)
+	if err != nil {
+		return nil, err
+	}
+	maxHeight := resp.SyncInfo.LatestBlockHeight
+
+	blockHeaders := make([]*types.BlockMeta, 0)
+	// fetch headers up to maxHeight
+	lastFetchedHeight := int64(0)
+	println("max height: ", maxHeight)
+	for {
+		// BlockchainInfo may apply a limit on the range of blocks to fetch,
+		// so we need to request them iteratively.
+		// note that block headers returned by BlockchainInfo are in descending
+		// order (highest first).
+		res, err := client.BlockchainInfo(ctx, 1, maxHeight)
+		if err != nil {
+			return nil, err
+		}
+
+		blockHeaders = append(blockHeaders, res.BlockMetas...)
+
+		lastFetchedHeight = res.BlockMetas[len(res.BlockMetas)-1].Header.Height
+
+		// fetch until the first block
+		if lastFetchedHeight <= 1 {
+			break
+		}
+
+		// set the new maxHeight to fetch the next batch of headers
+		maxHeight = lastFetchedHeight - 1
+
+	}
+
+	// reverse the order of headers to be ascending (lowest first).
+	reverseSlice(blockHeaders)
```

Comment:
[nit|personal opinion|feel free to ignore]
might be refactored to:

```suggestion
	blockchainInfoLimit := int64(20) // taken from: celestia-core/rpc/core/blocks.go: L32
	for startHeight := int64(1); startHeight < maxHeight; startHeight += blockchainInfoLimit {
		endHeight := startHeight + blockchainInfoLimit - 1
		res, err := client.BlockchainInfo(ctx, startHeight, endHeight)
		if err != nil {
			return nil, err
		}

		metas := res.BlockMetas
		reverseSlice(metas)
		blockHeaders = append(blockHeaders, metas...)
	}
	
	if int64(len(blockHeaders)) != maxHeight {
		// this in case we want all the headers, and we don't want to miss any. 
		// because BlockchainInfo might not return all the headers if the node is not synced or pruned
		// without the API returning an error.
		return nil, fmt.Errorf("couldn't fetch all the necessary headers")
	}
```

Reply:
Thank you for your suggestion. I initially intended to do it this way, but I realized the 20 block limit might not always apply. Therefore, I adopted an approach that isn't dependent on the hardcoded 20 block limit (to make it future proof).

Reply:
I see 👍